cmd/zoekt-sourcegraph-indexserver/main.go

```diff
@@ -184,12 +188,21 @@ type Server struct {
 
 var debug = log.New(io.Discard, "", log.LstdFlags)
 
-// our index commands should output something every 100mb they process.
-//
-// 2020-11-24 Keegan. "This should be rather quick so 5m is more than enough
-// time."  famous last words. A client was indexing a monorepo with 42
-// cores... 5m was not enough.
-const noOutputTimeout = 30 * time.Minute
+const (
+	// our index commands should output something every 100mb they process.
+	//
+	// 2020-11-24 Keegan. "This should be rather quick so 5m is more than enough
+	// time."  famous last words. A client was indexing a monorepo with 42
+	// cores... 5m was not enough.
+	noOutputTimeout = 30 * time.Minute
+
+	loggerScope = "zoekt-sourcegraph-indexserver"
```

Comment:
The scope should not be the service name, see https://docs.sourcegraph.com/dev/how-to/add_logging#handling-loggers

> The first top-level scoped logger is typically "server", since most logging is related to server initialization and service-level logging - the name of the service itself is already logged as part of the [Resource field](https://opentelemetry.io/docs/reference/specification/logs/data-model/#field-resource) provided during initialization.

Reply:
They should represent the subsystem _within_ `zoekt-sourcegraph-indexserver` that is most relevant:

> Scopes are used to identify the component of a system a log message comes from, and generally should provide enough information for an uninitiated reader (such as a new teammate, or a Sourcegraph administrator) to get a rough idea the context in which a log message might have occured.

Reply:
I see, that makes sense. I missed that we are duplicating the service name with both Scope and Resource referencing the system `zoekt-sourcegraph-indexserver`. I'll chat with the team re: what subsystem names make sense for these scopes

Reply:
Updated to use scope "server". @stefanhengl re: the name of service I added [`findName()`](https://github.com/sourcegraph/zoekt/pull/374/files#diff-2a5a3ba8904727b248161f4d3a99e9b7d717d50ab01d1149ebd2a8bbea9e7e4bR922) and I assign the result to the `Resource`. findName() is copied from sourcegraph/sourcegraph https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/internal/env/env.go?L55:6